h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/rabit/RabitWorker.java

```diff
@@ -0,0 +1,157 @@
+package hex.tree.xgboost.rabit;
+
+import hex.tree.xgboost.rabit.util.LinkMap;
+import water.AutoBuffer;
+import water.ExternalFrameUtils;
+import water.util.Log;
+
+import java.io.IOException;
+import java.nio.channels.SocketChannel;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class RabitWorker implements Comparable<RabitWorker> {
+
+    final String host;
+    private SocketChannel socket;
+    int rank;
+    int worldSize;
+    String jobId;
+    public String cmd;
+    int waitAccept;
+    private int port;
+    private AutoBuffer ab;
+    private AutoBuffer writerAB;
+
+
+    public RabitWorker(SocketChannel channel) throws IOException {
+        this.ab = new AutoBuffer(channel, null);
+        this.socket = channel;
+        this.host = channel.socket().getInetAddress().getHostAddress();
+        int magicReceived = ab.get4();
+        if(RabitTrackerH2O.MAGIC != magicReceived) {
+            throw new IllegalStateException(
+                    "Tracker received wrong magic number ["
+                    + magicReceived +
+                    "] from host " + this.host
```

Comment:
One proposal - it might be good to report also port so we can distinguish different trackers when running on the same physical node

Reply:
Yes I was thinking about that but the remote ports on Rabit workers (there's only 1 tracker, rest are workers) are being opened in Rabit's C++ code and there's no way (without changing the code) to print them out, so it wouldn't really help us :-(